plugins/com.google.cloud.tools.eclipse.appengine.libraries/src/com/google/cloud/tools/eclipse/appengine/libraries/model/Library.java

```diff
@@ -194,7 +194,14 @@ synchronized void setResolved() {
    * transitive dependency graph.
    */
   private synchronized List<LibraryFile> resolveDependencies() {
+    
+    if (transitiveDependencies != null) {
+      return transitiveDependencies; 
+    }
```

Comment:
Sensible optimization, but just letting you know this is redundant at the current state, because this method is called only when `transitiveDependencies != null` by `getAllDependencies()`.

Reply:
removed